### PR TITLE
add interactive frontend created with create-react-app

### PR DESCRIPTION
**Pull request by nichanank**
- barebones Home page and Project repo with functional `deposit` and `withdraw` features
- ability to connect, change and disconnect accounts
- tested on Kovan with deployed contracts
- install dependencies in the `client` and `react-app` folders and then from the `client` folder, run `yarn react-app:start` and it should spin up on localhost. make sure your MetaMask is on the right network (Kovan)
- NOTE: you have to be connected to an account to view the Projects page for now. It will fail on navigation if the wallet is not connected

### CODE REVIEW

**Review by Pongch — APPROVED** [review]
we should get this merged asap as well as try to deploy it to GitHub Page to see if things work correctly as expected

File: client/packages/react-app/public/index.html [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    <!--
+      Notice the use of %PUBLIC_URL% in the tags above.
+      It will be replaced with the URL of the `public` folder during the build.
+      Only files inside the `public` folder can be referenced from the HTML.
+
+      Unlike "/favicon.ico" or "favicon.ico", "%PUBLIC_URL%/favicon.ico" will
+      work correctly both with client-side routing and a non-root public URL.
+      Learn how to configure a non-root public URL by running `npm run build`.
+    -->
+    <title>Ethereum App</title>
```
Comment (jrhite) [inline client/packages/react-app/public/index.html:27]: want to change this and other references to Bazaar Finance?

File: client/packages/react-app/public/manifest.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{
+  "short_name": "Eth App",
```
Comment (jrhite) [inline client/packages/react-app/public/manifest.json:2]: change to Bazaar Finance references

File: client/packages/react-app/src/components/Web3ReactManager/index.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
+import React, { useState, useEffect } from 'react'
+import styled, { keyframes } from 'styled-components'
+import { useWeb3React } from '@web3-react/core'
+import { network } from '../../connectors'
```
Comment (jrhite) [inline client/packages/react-app/src/components/Web3ReactManager/index.js:4]: interesting...so we're using ethers on the contract side and hardhat. but using web3 for frontend? i'm guessing this was just a decision made in a CRA template?

File: hardhat.kovan-staging-config.ts [hunk 4]
```diff
@@ -45,7 +45,7 @@ const config: HardhatUserConfig = {
   abiExporter: {
     path: './data/abi',
     clear: true,
-    only: ['ILendingPool', 'IERC20', 'IVaultFactory', 'BazrToken', 'Vault', 'ProxyFactory']
+    only: ['ILendingPool', 'IERC20', 'VaultFactory', 'BazrToken', 'Vault', 'ProxyFactory']
```
Comment (jrhite) [inline hardhat.kovan-staging-config.ts:48]: 👍

**Review by jrhite — APPROVED** [review]
only minor comments. looks great!

File: client/packages/react-app/src/components/Web3ReactManager/index.js (hunk 3 above)
Comment (nichanank) [reply client/packages/react-app/src/components/Web3ReactManager/index.js:4]: web3React is just a library that abstracts low level web3 library functionality. we can specify which library we use, which in this case is still ethers. relevant lines are [here](https://github.com/bazaarfinance/bazaarfinance/blob/client/client/packages/react-app/src/index.js#L3) and [here](https://github.com/bazaarfinance/bazaarfinance/blob/client/client/packages/react-app/src/index.js#L18)

File: client/packages/react-app/src/utils/index.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    return library.getSigner(account).connectUnchecked();
+}
+
+export async function getEvents(library, filter) {
+    return library.getLogs(filter);
+}
+
+export async function getProjects(address, library, account) {
+    const vaultFactory = getVaultFactoryContract(address, library, account)
+    const filter = vaultFactory.filters.VaultCreated(null, null);
```
[the remark this answers is not in the document]
Comment (nichanank) [reply client/packages/react-app/src/utils/index.js:31]: yep, can use the abi to filter events in addition to calling contract functions

**Pull request merged**